textproto/header_test.go

```diff
@@ -230,29 +230,21 @@ func TestInvalidHeader(t *testing.T) {
 	}
 }
 
-func TestReadHeader_TooBig(t *testing.T) {
+func TestReadHeader_LimitReader(t *testing.T) {
 	testHeader := "Received: from example.com by example.org\r\n" +
 		"Received: from localhost by example.com\r\n" +
 		"To: Taki Tachibana <taki.tachibana@example.org> " + strings.Repeat("A", 4000) + "\r\n" +
 		"From: Mitsuha Miyamizu <mitsuha.miyamizu@example.com>\r\n\r\n"
-	_, err := ReadHeader(bufio.NewReader(strings.NewReader(testHeader)))
-	if err == nil {
-		t.Fatalf("ReadHeader() succeeded")
+	h, err := ReadHeader(bufio.NewReader(io.LimitReader(strings.NewReader(testHeader), 140)))
```

Comment:
I'm not sure what this is testing.

Reply:
updated test case, thanks for the comment.